tequilapi/endpoints/transactor.go

```diff
@@ -292,8 +298,7 @@ func (te *transactorEndpoint) SettleWithBeneficiary(resp http.ResponseWriter, re
 		return
 	}
 
-	chainID := config.GetInt64(config.FlagChainID)
-	err = te.promiseSettler.SettleWithBeneficiary(chainID, identity.FromAddress(id), common.HexToAddress(req.Beneficiary), common.HexToAddress(req.HermesID))
```

Comment:
This is incorrect. In both cases, we want to settle with beneficiary. 

In the case of l1 beneficiary settle, we settle with the beneficiary address provided by the user. This automatically serves as a save.

In the case of l2 beneficiary settle, we settle with the generated consumer channel address and save the provided address in the database.

While getting beneficiary addresses, for l2 we want to fetch it from db, for l1 we want to fetch it from BC.

Reply:
Perhaps we can leave the settlement here, and use the beneficiary handlers that you've made, except change the save method on the l1 handler to just do nothing.

Reply:
You're right I missed that somehow. Totally my bad here.